src/wazuh_modules/inventory_harvester/src/systemInventory/elements/netIfaceElement.hpp

```diff
@@ -0,0 +1,100 @@
+/*
+ * Wazuh Inventory Harvester - Network Interface element
+ * Copyright (C) 2015, Wazuh Inc.
+ * Match 25, 2025.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#ifndef _NETIFACE_ELEMENT_HPP
+#define _NETIFACE_ELEMENT_HPP
+
+#include "../../wcsModel/data.hpp"
+#include "../../wcsModel/inventoryNetIfaceHarvester.hpp"
+#include "../../wcsModel/noData.hpp"
+#include "stringHelper.h"
+#include <loggerHelper.h>
+
+template<typename TContext>
+class NetIfaceElement final
+{
+public:
+    // LCOV_EXCL_START
+    /**
+     * @brief Class destructor.
+     *
+     */
+    ~NetIfaceElement() = default;
+    // LCOV_EXCL_STOP
+
+    static DataHarvester<InventoryNetIfaceHarvester> build(TContext* data)
+    {
+        auto agentId = data->agentId();
+        if (agentId.empty())
+        {
+            throw std::runtime_error("Agent ID is empty");
+        }
+
+        auto netIfaceItemId = data->netIfaceItemId();
```

Comment:
comment: Probably in the future, we can re-use this common functions, at least for network messages.

Reply:
Sure, we need to make review of everything after the components implementation to unify.